hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/MiniDFSCluster.java

```diff
@@ -2775,14 +2775,23 @@ public void waitActive(int nnIndex) throws IOException {
     DFSClient client = new DFSClient(addr, conf);
 
     // ensure all datanodes have registered and sent heartbeat to the namenode
-    while (shouldWait(client.datanodeReport(DatanodeReportType.LIVE), addr)) {
-      try {
+    int failedCount = 0;
+    try {
+      while (shouldWait(client.datanodeReport(DatanodeReportType.LIVE), addr)) {
         LOG.info("Waiting for cluster to become active");
         Thread.sleep(100);
-      } catch (InterruptedException e) {
       }
+    } catch (IOException e) {
+      failedCount++;
+      // Cached RPC connection to namenode, if any, is expected to fail once
+      if (failedCount > 1) {
```

Comment:
Now that we have kept entire while loop within try-catch, `failedCount` will be `1` here due to `failedCount++` and then we exit from try-catch so `failedCount` will never be recalculated. Recalculation will be possible only if try-catch is kept within while loop. Based on WARN log message, it seems that's what we should do because we are trying to add retries?

Reply:
the while looped pulled in should solve, else I need to spend some time during this weekend. 
my Holidays ends today. :-)